api/client/cli.go

```diff
@@ -106,6 +107,12 @@ func (cli *DockerCli) restoreTerminal(in io.Closer) error {
 	return nil
 }
 
+// AddCommonFlags installs command line flags which can control the
+// client's behavior, and which are common to all client commands.
+func (cli *DockerCli) AddCommonFlags(flags *flag.FlagSet) {
+	// nothing here yet
```

Comment:
I'd be all for moving some subset of these into here:
```
  --config=~/.docker                 Location of client config files
  -D, --debug=false                  Enable debug mode
  --disable-legacy-registry=false    Do not contact legacy registries
  -H, --host=[]                      Daemon socket(s) to connect to
  -h, --help=false                   Print usage
  -l, --log-level=info               Set the logging level
  --tls=false                        Use TLS; implied by --tlsverify
  --tlscacert=~/.docker/ca.pem       Trust certs signed only by this CA
  --tlscert=~/.docker/cert.pem       Path to TLS certificate file
  --tlskey=~/.docker/key.pem         Path to TLS key file
  --tlsverify=false                  Use TLS and verify the remote
```
It's really annoying to do stuff like `docker run -D ...` and get `flag provided but not defined: -D` because I needed to do `docker -D run`